erts/emulator/beam/beam_load.c

```diff
@@ -6332,7 +6355,7 @@ erts_make_stub_module(Process* p, Eterm Mod, Eterm Beam, Eterm Info)
 	goto error;
     }
     define_file(stp, "atom table", ATOM_CHUNK);
-    if (!load_atom_table(stp)) {
+    if (!load_atom_table(stp, ERTS_ATOM_ENC_LATIN1)) {
```

Comment:
You'll need to go through the same `if (stp->chunks[UTF8_ATOM_CHUNK].size > 0) ...` charade down here, or `code:make_stub_module/3` (and thus HiPE) will break. There's some tests for it in `erts/emulator/test/code_SUITE*`